junit-tests/src/test/java/org/junit/gen5/console/tasks/TempDirectory.java

```diff
@@ -71,7 +71,15 @@ private Namespace localNamespace(ExtensionContext context) {
 
 	private Path createTempDirectory(ExtensionContext context) {
 		try {
-			return Files.createTempDirectory(context.getName());
+			String tempDirName;
+			if (context.getTestMethod().isPresent()) {
+				tempDirName = context.getTestMethod().get().getName();
+			}
+			else {
+				tempDirName = context.getTestClass().get().getName();
```

Comment:
This call to `get()` strikes me as not being save in all cases. Can we use an additional fallback if it the test class is not present, e.g. `getDisplayName()`?